packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+                        ? `tab-${index + 1}`
+                        : this.tabIDs[index]
+                );
+                tab.setAttribute(
+                    "aria-selected",
+                    this.activeTabIndex === index ? "true" : "false"
+                );
+                tab.setAttribute(
+                    "aria-controls",
+                    this.tabPanelIDs[index] === undefined ||
+                        this.tabPanelIDs[index] === null
+                        ? `panel-${index + 1}`
+                        : this.tabPanelIDs[index]
+                );
+                tab.setAttribute(
+                    "style",
+                    this.isHorizontal()
+                        ? `grid-column: ${index + 1};`
+                        : `grid-row: ${index + 1};`
+                );
+                tab.addEventListener("click", this.handleTabClick);
+                tab.addEventListener("keydown", this.handleTabKeyDown);
+                tab.setAttribute("tabindex", this.activeTabIndex === index ? "0" : "-1");
+                if (this.activeTabIndex === index) {
+                    this.activeTab = tab;
+                }
+                !this.isHorizontal()
+                    ? tab.classList.add("vertical")
+                    : tab.classList.remove("vertical");
+            }
+        });
+    };
+
+    private getTabPanels = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
+        this.tabPanels.forEach((tabpanel: HTMLElement, index: number) => {
+            tabpanel.setAttribute(
+                "id",
+                this.tabPanelIDs[index] === undefined || this.tabPanelIDs[index] === null
+                    ? `panel-${index + 1}`
+                    : this.tabPanelIDs[index]
+            );
+            tabpanel.setAttribute(
+                "aria-labeledby",
+                this.tabIDs[index] === undefined || this.tabIDs[index] === null
+                    ? `tab-${index + 1}`
+                    : this.tabIDs[index]
+            );
+            this.activeTabIndex !== index
+                ? tabpanel.setAttribute("hidden", "")
+                : tabpanel.removeAttribute("hidden");
+        });
+    };
+
+    private getTabIds(): string[] {
+        return this.tabs.map((tab: HTMLElement) => {
+            return tab.getAttribute("id") as string;
+        });
+    }
+
+    private getTabPanelIds(): string[] {
+        return this.tabPanels.map((tabPanel: HTMLElement) => {
+            return tabPanel.getAttribute("id") as string;
+        });
+    }
+
+    private setTabs(): void {
+        this.activeId = this.tabIDs[this.activeTabIndex];
+        this.change();
+        this.getTabs();
+        this.handleActiveIndicatorPosition();
+        this.getTabPanels();
+        this.focusTab();
+        this.change();
+    }
+
+    private handleTabClick = (event): void => {
+        const selectedTab = event.srcElement as Element;
```

Comment:
you shouldn't use `srcElement` - it is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/Event/srcElement